chore: add a security policy (#1184)
ApeWorX/ape: SECURITY.md

Any expected deviations and necessary clarifications around the standard are explained in the following sections.

## Disclosing a Vulnerability

### Supported Versions

Ape is currently under active development, targeting a 1.x release.
Any security issues found on the latest minor version of Ape will be fixed and released in either a patch or minor release.

| Version | Supported          |
| ------- | ------------------ |
| > 0.5.x | :white_check_mark: |

### Directly to ApeWorX

ApeWorX is committed to working with researchers who submit security vulnerability notifications to us, to resolve those issues on an appropriate timeline, and to perform a coordinated release, giving credit to the reporter if they would so like.

Please submit disclosures to **all** of the following main points of contact for security related issues according to the [initial contact](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#initial-contact) and [giving details](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#giving-details) guidelines.

For all security related issues, ApeWorX has the following main points of contact:

> | Contact                                    | Keybase                                         |
> | ------------------------------------------ | ----------------------------------------------- |
> | [@fubuloubu](https://github.com/fubuloubu) | [@fubuloubu](https://keybase.io/fubuloubu/chat) |

Author: ```suggestion
| Contact                                              | Keybase                                                   |
| ---------------------------------------------------- | --------------------------------------------------------- |
| [@fubuloubu](https://github.com/fubuloubu)           | [@fubuloubu](https://keybase.io/fubuloubu/chat)           |
| [@NotPeopling2day](https://github.com/NotPeopling2day) | [@NotPeopling2day](https://keybase.io/notepeopling2day/chat) |
```

> # Security Policy

Reviewer: Should we link to anything official regarding what this is? Perhaps https://csrc.nist.gov/glossary/term/security_policy

> This document describes the Security Policy for ApeWorX Ltd., including vulnerability disclosures for the [Ape Framework](https://github.com/ApeWorX/ape) and any of it's [maintained plugins](https://github.com/search?q=org%3Aapeworx+ape-+-repo%3AApeWorX%2Fape++-repo%3AApeWorX%2Fapeworx.github.io&type=Repositories).

Reviewer: I just realized - should we have a policy that includes all repos we maintain, including `ethpm-types` and all that? If so, where would that live?

Author: Could add that here I suppose

Author: ```suggestion
This document describes the Security Policy for ApeWorX Ltd., including vulnerability disclosures for the Ape Framework, any of it's maintained plugins, or any packages we maintain used by those as dependencies (see [here](](https://github.com/search?q=org%3Aapeworx+-repo%3AApeWorX%2Fapeworx.github.io&type=Repositories))).
```

We are committed to conduct our Security Process in a professional and civil manner.

> Public shaming, under-reporting, or misrepresentation of vulnerabilities will not be tolerated.

Reviewer: Are there consequences? What does this mean?

Author: I don't know, might be more specific to smart contract exploits

To submit a finding, please follow the steps outlined in receiving disclosures [section](#disclosing-a-vulnerability).

## Responsible Disclosure Standard

ApeWorX follows a community [standard](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#the-standard) for responsible disclosure in cryptocurrency and related software.
This document is a public commitment to following the standard.

This standard provides detailed information for:

- [Initial Contact](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#initial-contact):
  how to establish initial contact with our security team.
- [Giving Details](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#giving-details):
  what details to include with your vulnerability disclosure after having received a response to your initial contact.
- [Setting Dates](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#setting-dates):
  how to agree on timelines for releasing updates and making details of the issue public.

Any expected deviations and necessary clarifications around the standard are explained in the following sections.

## Disclosing a Vulnerability

### Supported Versions

> Ape is currently under active development, targeting a 1.x release.

Reviewer: Is there any way to say this (and feel free to disagree) without having to remember to come back and update it once we hit 1.0?

Author: Maybe
```suggestion
Ape is currently under active development, we support the latest release by major version.
```

Any security issues found on the latest minor version of Ape will be fixed and released in either a patch or minor release.

| Version | Supported          |
| ------- | ------------------ |
| > 0.5.x | :white_check_mark: |

### Directly to ApeWorX

ApeWorX is committed to working with researchers who submit security vulnerability notifications to us, to resolve those issues on an appropriate timeline, and to perform a coordinated release, giving credit to the reporter if they would so like.

Please submit disclosures to **all** of the following main points of contact for security related issues according to the [initial contact](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#initial-contact) and [giving details](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#giving-details) guidelines.

For all security related issues, ApeWorX has the following main points of contact:

> | Contact                                                | Keybase                                                     |

Author: ```suggestion
| Contact                                                | Keybase                                                      |
```

> | Contact                                              | Keybase                                                   |
> | ---------------------------------------------------- | --------------------------------------------------------- |
> | [@fubuloubu](https://github.com/fubuloubu)           | [@fubuloubu](https://keybase.io/fubuloubu/chat)           |

Author: ```suggestion
| Contact                                               | Keybase                                                    |
| ----------------------------------------------------- | ----------------------------------------------------------- |
| [@fubuloubu](https://github.com/fubuloubu)            | [@fubuloubu](https://keybase.io/fubuloubu/chat)             |
```

> | Contact                                               | Keybase                                                    |
> | ----------------------------------------------------- | ----------------------------------------------------------- |
> | [@fubuloubu](https://github.com/fubuloubu)            | [@fubuloubu](https://keybase.io/fubuloubu/chat)             |

Author: ```suggestion
| Contact                                                | Keybase                                                     |
| ------------------------------------------------------ | ------------------------------------------------------------ |
| [@fubuloubu](https://github.com/fubuloubu)             | [@fubuloubu](https://keybase.io/fubuloubu/chat)              |
```

> | [@NotPeopling2day](https://github.com/NotPeopling2day) | [@NotPeopling2day](https://keybase.io/notepeopling2day/chat) |

Reviewer: I feel left out. Do we need more people here? I volunteer if yes...

Author: Yes, but you need to add yourself to Keybase lol

Reviewer: My username is `antazoey`!

> | Contact                                                | Keybase                                                      |
> | ------------------------------------------------------ | ------------------------------------------------------------ |
> | [@fubuloubu](https://github.com/fubuloubu)             | [@fubuloubu](https://keybase.io/fubuloubu/chat)              |
> | [@NotPeopling2day](https://github.com/NotPeopling2day) | [@NotPeopling2day](https://keybase.io/notepeopling2day/chat) |

Author: ```suggestion
| Contact                                                | Keybase                                                      |
| ------------------------------------------------------ | ------------------------------------------------------------ |
| [@fubuloubu](https://github.com/fubuloubu)             | [@fubuloubu](https://keybase.io/fubuloubu/chat)              |
| [@NotPeopling2day](https://github.com/NotPeopling2day) | [@NotPeopling2day](https://keybase.io/notpeopling2day/chat) |
| [@unparalleled-js](https://github.com/unparalleled-js) | [@Antazoey](https://keybase.io/antazoey/chat) |
```

Any expected deviations and necessary clarifications around the standard are explained in the following sections.

## Disclosing a Vulnerability

### Supported Versions

Ape is currently under active development, targeting a 1.x release.
Any security issues found on the latest minor version of Ape will be fixed and released in either a patch or minor release.

| Version | Supported          |
| ------- | ------------------ |
| > 0.5.x | :white_check_mark: |

### Directly to ApeWorX

ApeWorX is committed to working with researchers who submit security vulnerability notifications to us, to resolve those issues on an appropriate timeline, and to perform a coordinated release, giving credit to the reporter if they would so like.

Please submit disclosures to **all** of the following main points of contact for security related issues according to the [initial contact](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#initial-contact) and [giving details](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#giving-details) guidelines.

For all security related issues, ApeWorX has the following main points of contact:

> | Contact                                                | Keybase                                                      |
> | ------------------------------------------------------ | ------------------------------------------------------------ |
> | [@fubuloubu](https://github.com/fubuloubu)             | [@fubuloubu](https://keybase.io/fubuloubu/chat)              |
> | [@NotPeopling2day](https://github.com/NotPeopling2day) | [@NotPeopling2day](https://keybase.io/notpeopling2day/chat) |
> | [@unparalleled-js](https://github.com/unparalleled-js) | [@Antazoey](https://keybase.io/antazoey/chat) |

Author: ```suggestion
| Contact                                                | Keybase                                                     |
| ------------------------------------------------------ | ----------------------------------------------------------- |
| [@fubuloubu](https://github.com/fubuloubu)             | [@fubuloubu](https://keybase.io/fubuloubu/chat)             |
| [@NotPeopling2day](https://github.com/NotPeopling2day) | [@NotPeopling2day](https://keybase.io/notpeopling2day/chat) |
| [@unparalleled-js](https://github.com/unparalleled-js) | [@Antazoey](https://keybase.io/antazoey/chat)              |
```

> | [@unparalleled-js](https://github.com/unparalleled-js) | [@Antazoey](https://keybase.io/antazoey/chat)              |

Author: ```suggestion
| [@unparalleled-js](https://github.com/unparalleled-js) | [@Antazoey](https://keybase.io/antazoey/chat)               |
```

# Security Policy

This document describes the Security Policy for ApeWorX Ltd.,
including vulnerability disclosures for the [Ape Framework](https://github.com/ApeWorX/ape)

> and any of it's [maintained plugins](https://github.com/ApeWorX?q=%22ape-%22&type=public&language=&sort=).

Reviewer: I don't really have a better answer yet, nor is this a huge deal, but this query is not 100% right. It shows a few other repos besides plugins in it.

Reviewer 2: Still could be improved https://github.com/search?q=org%3Aapeworx+ape-&type=repositories

Reviewer: For some reason, the docs site repo shows in that list

Author: Found a better one

ApeWorX follows a community [standard](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#the-standard)
for responsible disclosure in cryptocurrency and related software.
This document is a public commitment to following the standard.

This standard provides detailed information for:

- [Initial Contact](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#initial-contact):
  how to establish initial contact with our security team.
- [Giving Details](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#giving-details):
  what details to include with your vulnerability disclosure after having received a response to your initial contact.
- [Setting Dates](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#setting-dates):
  how to agree on timelines for releasing updates and making details of the issue public.

Any expected deviations and necessary clarifications around the standard are explained in the following sections.

## Disclosing a Vulnerability

### Supported Versions

Ape is currently under active development, targeting a 1.x release.
Any security issues found on the latest minor version of Ape will be fixed and released in either

> a patch or minor release.

Reviewer: The formatting of this document is veering from our norm of always doing 1 line per sentence in markdown.

| Version | Supported          |
| ------- | ------------------ |
| > 0.5.x | :white_check_mark: |

### Directly to ApeWorX

ApeWorX is committed to working with researchers who submit security vulnerability notifications to us,
to resolve those issues on an appropriate timeline, and to perform a coordinated release,
giving credit to the reporter if they would so like.

Please submit disclosures to **all** of the following main points of contact for
security related issues according to the
[initial contact](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#initial-contact)
and [giving details](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#giving-details)
guidelines.

For all security related issues, ApeWorX has the following main points of contact:

| Contact                                    | Keybase                                         |
| ------------------------------------------ | ----------------------------------------------- |
| [@fubuloubu](https://github.com/fubuloubu) | [@fubuloubu](https://keybase.io/fubuloubu/chat) |

## Sending Disclosures

In the case where we become aware of security issues affecting projects using Ape,

> our intention is to engage in responsible disclosures with them as described in the adopted 

Reviewer: ```suggestion
our intention is to engage in responsible disclosures with them as described in the adopted
```
